### PR TITLE
Bug eci 222 add abandoned cart test

### PR DESCRIPTION
**Pull request by starshine1510**
Ported over the abandoned cart test to M2 since we added it to M1.
Abandoned cart url doesn't seem to work the same as M1.
Visiting an abandoned cart url takes you to an empty shopping cart.
Not sure if this is due to how M2 handles sessions or if we are manipulating the cart url incorrectly.
We should look into why but for now the test is in place even though I don't think the results are what we really want.  I'm out of ideas at this point.

### CODE REVIEW

File: devtools_m2/cypress/integration/CustomerCartInteractions/steps.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ When('I add a {string} widget to my cart', function(type) {
       break;
     case 'simple':
       // Do nothing
+      console.log('In simple case')
```
Comment (will-in-wi) [inline devtools_m2/cypress/integration/CustomerCartInteractions/steps.js]: Could we drop the console.log or convert it to https://docs.cypress.io/api/commands/log.html#Syntax ?

File: devtools_m2/cypress/integration/CustomerCartInteractions/steps.js [hunk 2]
```diff
@@ -136,6 +139,7 @@ Then('A simple cart event should be sent to Drip', function() {
     expect(item.quantity).to.eq(1)
     expect(item.total).to.eq(11.22)
   })
+
```
Comment (will-in-wi) [inline devtools_m2/cypress/integration/CustomerCartInteractions/steps.js]: I really want to add a proper linter to these files. Pending that, could this newline be deleted?

**Review by will-in-wi — CHANGES_REQUESTED** [review]
Could you delete the `devtools_m2/git` file? It doesn't seem needed.

I agree that this test is a net positive even if we don't have the same functionality in M1 and M2. However, I'd really like to have us dig into this, or at least throw a ticket onto the backlog to do so. I'm a little nervous that we're going to have a customer upgrade from M1 to M2 and then be surprised when we don't have this feature.

**Review by tdieterich-drip — APPROVED** [review]

**Comment by iantance** [comment]
@starshine1510 Looks like we were attempting to build the cart by clearing the current cart's item and then merging the items from their abandoned cart. But when the current cart _is_ the abandoned cart, we end up with an empty cart.

**Review by will-in-wi — APPROVED** [review]

**Pull request merged**